src/ufoLib2/objects/layer.py

```diff
@@ -203,6 +209,44 @@ def name(self) -> str:
         """The name of the layer."""
         return self._name
 
+    @property
+    def bounds(self) -> Optional[BoundingBox]:
+        """Returns the (xMin, yMin, xMax, yMax) bounding box of the layer,
+        taking the actual contours into account.
+
+        |defcon_compat|
+        """
+        bounds = None
+        for glyph in self:
+            glyphBounds = glyph.getBounds(self)
+            if glyphBounds is None:
+                continue
+            if bounds is None:
+                bounds = glyphBounds
+            else:
+                bounds = unionBounds(bounds, glyphBounds)
```

Comment:
It is probably too late for me to think properly, but I have absolutely no idea why mypy thinks this line is unreachable.